### PR TITLE
Add an overload to ResolutionExtensions.InjectProperties to accept IPropertySelector

### LINKED ISSUE #761: Add overload to InjectProperties to accept IPropertySelector
`IPropertySelector` was recently added (#715) to allow for selective property injection with `IRegistrationBuilder.PropertiesAutowired`. There is another entry point for people to inject properties via the `IComponentContext.InjectProperties` method. I propose to add an overload that accepts IPropertySelector:

```csharp
 public static TService InjectProperties<TService>(this IComponentContext context, TService instance, IPropertySelector propertySelector);
```

### PR DESCRIPTION
**Pull request by twsouthwick**
This fixes #761

### CODE REVIEW

**Comment by twsouthwick** [comment]
I also added cached DefaultPropertySelector instances and used them where possible. I can remove that if it's not wanted.

**Comment by twsouthwick** [comment]
Build is failing due to Autofac.Extensions.DependencyInjection.Test restore - unrelated to this change

**Comment by tillig** [comment]
Yeah, there was an xunit beta update that broke stuff. I just pushed a fix.

File: src/Autofac/Core/Activators/DefaultPropertySelector.cs [hunk 1]
```diff
@@ -11,6 +11,16 @@ namespace Autofac.Core
     public class DefaultPropertySelector : IPropertySelector
     {
         /// <summary>
+        /// Gets an instance of DefaultPropertySelector that will preserve an values already set
```
Comment (tillig) [inline src/Autofac/Core/Activators/DefaultPropertySelector.cs]: Minor typo - "any"

File: src/Autofac/ResolutionExtensions.cs [hunk 2]
```diff
@@ -55,7 +55,22 @@ public static class ResolutionExtensions
         /// <returns><paramref name="instance"/>.</returns>
         public static TService InjectProperties<TService>(this IComponentContext context, TService instance)
         {
-            AutowiringPropertyInjector.InjectProperties(context, instance, true);
+            AutowiringPropertyInjector.InjectProperties(context, instance, DefaultPropertySelector.OverwriteSetValueInstance);
+            return instance;
+        }
+
+        /// <summary>
+        /// Set any null-valued properties on <paramref name="instance"/> that can be
```
Comment (tillig) [inline src/Autofac/ResolutionExtensions.cs]: I think the summary here may be a copy/paste error. It uses the selector, not null-value properties.

**Comment by tillig** [comment]
Looks good and straightforward. A couple of very minor doc updates and we're good to go.

File: src/Autofac/ResolutionExtensions.cs (hunk 2 above)
Comment (twsouthwick) [inline src/Autofac/ResolutionExtensions.cs]: Yup - didn't catch the summary. Fixed

File: src/Autofac/Core/Activators/DefaultPropertySelector.cs (hunk 1 above)
Comment (twsouthwick) [inline src/Autofac/Core/Activators/DefaultPropertySelector.cs]: :)

**Comment by tillig** [comment]
Rock and roll. Thanks again for the help!

**Comment by twsouthwick** [comment]
Thanks for taking it!

**Pull request merged**